src/test/java/seedu/anilist/model/anime/EpisodeTest.java

```diff
@@ -14,29 +26,36 @@ public void constructor_null_throwsNullPointerException() {
 
     @Test
     public void constructor_invalidEpisode_throwsIllegalArgumentException() {
-        String invalidEpisode = "";
-        assertThrows(IllegalArgumentException.class, () -> new Episode(invalidEpisode));
+        assertThrows(IllegalArgumentException.class, () -> new Episode(INVALID_STRING_SPACE));
+        assertThrows(IllegalArgumentException.class, () -> new Episode(INVALID_STRING_EMPTY));
+        assertThrows(IllegalArgumentException.class, () -> new Episode(INVALID_STRING_NON_ASCII));
+        assertThrows(IllegalArgumentException.class, () -> new Episode(INVALID_EPISODE_ALPHA));
+        assertThrows(IllegalArgumentException.class, () -> new Episode(INVALID_EPISODE_DECIMAL));
+        assertThrows(IllegalArgumentException.class, () -> new Episode(INVALID_EPISODE_NEG));
+        assertThrows(IllegalArgumentException.class, () -> new Episode(INVALID_EPISODE_LARGER_THAN_MAX_EPISODE));
+        assertThrows(IllegalArgumentException.class, () -> new Episode(INVALID_EPISODE_LARGER_THAN_MAX_INT));
     }
 
     @Test
     public void isValidEpisode() {
-        // null name
+        // null episode
         assertThrows(NullPointerException.class, () -> Episode.isValidEpisode(null));
 
-        // invalid name
-        assertFalse(Episode.isValidEpisode("")); // empty string
-        assertFalse(Episode.isValidEpisode(" ")); // spaces only
-        assertFalse(Episode.isValidEpisode("^")); // only non-numeric characters
-        assertFalse(Episode.isValidEpisode("^123%")); // contains non-numeric characters
-        assertFalse(Episode.isValidEpisode("-1")); // negative numbers
-        assertFalse(Episode.isValidEpisode("3.0")); // float numbers
-        assertFalse(Episode.isValidEpisode(Long.toString((long) Integer.MAX_VALUE + 1))); // over maximum value
-
-        // valid name
-        assertTrue(Episode.isValidEpisode("12345")); // numbers only
-        assertTrue(Episode.isValidEpisode("0000")); // leading zeroes
-
-        // larger than 99999
-        assertFalse(Episode.isValidEpisode(Integer.toString(Integer.MAX_VALUE))); // maximum value
+        // invalid episode
+        assertFalse(Episode.isValidEpisode(INVALID_STRING_SPACE));
+        assertFalse(Episode.isValidEpisode(INVALID_STRING_EMPTY));
+        assertFalse(Episode.isValidEpisode(INVALID_STRING_NON_ASCII));
+        assertFalse(Episode.isValidEpisode(INVALID_EPISODE_ALPHA));
+        assertFalse(Episode.isValidEpisode(INVALID_EPISODE_DECIMAL));
+        assertFalse(Episode.isValidEpisode(INVALID_EPISODE_NEG));
+        assertFalse(Episode.isValidEpisode(INVALID_EPISODE_LARGER_THAN_MAX_EPISODE));
+        assertFalse(Episode.isValidEpisode(INVALID_EPISODE_LARGER_THAN_MAX_INT));
+
+        // valid episodes
+        assertTrue(Episode.isValidEpisode(VALID_EPISODE_ZERO));
+        assertTrue(Episode.isValidEpisode(VALID_EPISODE_ONE));
+        assertTrue(Episode.isValidEpisode(VALID_EPISODE_TWO));
+        assertTrue(Episode.isValidEpisode(VALID_EPISODE_MAX));
```

Comment:
probably can reduce the test cases here

Reply:
reduced the invalid strings, as for the invalid numbers i think we should still include them